### PR TITLE
Add Oberon-07 implementations

### PR DESCRIPTION
**Pull request by johnperry-math**
I've written a couple of Oberon-07 implementations. One works on the Java Virtual Machine, and performs about as fast as the Java implementation (a tad slower); the other works natively, and has a timing comparable to the C++ implementation.

### CODE REVIEW

**Comment by johnperry-math** [comment]
I neglected to mention that the good timing for the native version (obnc compiler) comes when you set `CFLAGS=-03`.

**Comment by frol** [comment]
@johnperry-math Can you update the README?

**Comment by johnperry-math** [comment]
I just pushed such a change. Do I need to open a new pull request? GitHub has a message here that suggests I don't.

**Comment by johnperry-math** [comment]
Actually, let me update the readme once more: I'll add direct links to the compiler websites.

File: oberon07/README.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+Then compile:
+
+```
+obnc CompleteUnscientificBenchmark.obn
+```
+
+**Do not** delete `Time.c`. This is necessary to interface to the C
+standard library to obtain the time, which we use to see the
+random number generator.
```
Comment (frol) [inline oberon07/README.md]: Well, this significantly complicates the solution. I think it is fine to use a constant random seed. What do you think?

**Comment by johnperry-math** [comment]
It's pretty normal for a lot of languages to interface with C, and I understand that the various revisions of Oberon have historically been among them, so in a way I think it's a feature. In fact, I'm not sure how we'd get a constant random seed otherwise? I thought using system time was standard.

**Comment by frol** [comment]
I see, OK. I don't have time to work on getting Oberon-07 on my machine now, so I cannot promise that I will merge it soon.

**Comment by johnperry-math** [comment]
No problem; I'm in no hurry. :-)

**Comment by frol** [comment]
I have rebased and squashed your commits into f50dd6497ee1e0b9f710861d83bd27f62b505608 and afdfe71c3166bfbed1e0acd38bcbff3ede14cf5d, and also have run benchmarks on my Linux machine and added the results for Modula-2, Modula-3, and Oberon-07 to the scoreboards!

**Comment by johnperry-math** [comment]
Awesome -- I thought this was a lost cause. Thank you so much!

**Comment by frol** [comment]
@johnperry-math I kept track of this PR and had a daily reminder, but it was quite a cumbersome process to install all those compilers (in fact, it was not that easy to even find them...), which was a major roadblocker as I didn't want to blindly add solutions without adding them to the scoreboards. Finally, I sorted this out! Thanks for your patience

**Comment by johnperry-math** [comment]
Thank you again. I'm aware that the compilers are a pain; I installed the Gnu Modula-2 compiler onto one of my Linux systems, and the other refuses to take it. They're great languages, and it's a shame to see them disappear.

**Pull request closed without merging**